src/components/Autocomplete/index.tsx

```diff
@@ -282,7 +282,7 @@ class AutocompleteInt<T> extends React.Component<AutocompleteProps<T> & FormCont
       inputRef: _7,
       ...props
     } = this.props;
-    const { open, listFocus, value, error } = this.state;
+    const { open, listFocus, value, error = this.props.error } = this.state;
```

Comment:
In that and other similar cases logic should be moved to `componentWillReceiveProps`.
Be careful: currently it's updating state only if component is in controlled mode. Instead it should update `error` on each prop change (not only in controlled mode)

Reply:
Check it now.